PSKoans/Koans/dbatools/AboutBackupDatabases.ps1

```diff
@@ -0,0 +1,167 @@
+
+            [PSCustomObject]@{
+                ComputerName = 'localhost'
+                SqlInstance = $ENV:COMPUTERNAME
+                Database = ('Database_{0:d2}' -f $_)
+                Type = 'Full'
+                TotalSize = ('{0}' -f ([Math]::PI * $_))
+                DeviceType = 'Disk'
+                Start = $RestoreStart
+                Duration = $RestoreEnd - $RestoreStart
+                End = $RestoreEnd
+            }
+
+            $StartDate = $RestoreEnd
+        }
+    } -ParameterFilter {
+        $_.Database -contains ('Database_01', 'Database_02')
+    }
+    Mock -CommandName Backup-DbaDatabase -MockWith {
+        $StartDate = [Datetime]::new(2019, 01, 01, 12, 00, 00)
+        $RestoreEnd = $StartDate.AddSeconds(5)
+        [PSCustomObject]@{
+            ComputerName = 'localhost'
+            SqlInstance = $ENV:COMPUTERNAME
+            Database = 'Database_01'
+            Type = 'Differential'
+            TotalSize = ('{0}' -f ([Math]::PI))
+            DeviceType = 'Disk'
+            Start = $StartDate
+            Duration = $RestoreEnd - $StartDate
+            End = $RestoreEnd
+        }
+    } -ParameterFilter {
+        $_.Type -eq 'Differential'
+    }
+    Mock -CommandName Backup-DbaDatabase -MockWith {
+        $StartDate = [Datetime]::new(2019, 01, 01, 12, 00, 00)
+        $RestoreEnd = $StartDate.AddSeconds(5)
+        [PSCustomObject]@{
+            ComputerName = 'localhost'
+            SqlInstance = $ENV:COMPUTERNAME
+            Database = 'Database_01'
+            Type = 'Differential'
+            TotalSize = ('{0}' -f ([Math]::PI))
+            DeviceType = 'Disk'
+            Start = $StartDate
+            Duration = $RestoreEnd - $StartDate
+            End = $RestoreEnd
+            Path = 'E:\Backups\Database_01_201901011200.bak'
+        }
+    } -ParameterFilter {
+        $_.Path -like 'E:\Backups\*'
+    }
+    Mock -CommandName Backup-DbaDatabase -MockWith {
+        $StartDate = [Datetime]::new(2019, 01, 01, 12, 00, 00)
+        $RestoreEnd = $StartDate.AddSeconds(5)
+        [PSCustomObject]@{
+            ComputerName = 'localhost'
+            SqlInstance = $ENV:COMPUTERNAME
+            Database = 'Database_01'
+            Type = 'Differential'
+            TotalSize = ('{0}' -f ([Math]::PI))
+            DeviceType = 'Disk'
+            Start = $StartDate
+            Duration = $RestoreEnd - $StartDate
+            End = $RestoreEnd
+            Path = 'E:\Backups\Database01-Full.bak'
+        }
+    } -ParameterFilter {
+        $_.FilePath -eq 'dbname-backuptype.bak'
+    }
+
+   
+
+    <#
+        By default, Backup-DbaDatabase will backup every database on the SQL Instance.
+        These backups will get saved to the default backup directory.
+        Complete the below command to backup all the databases on the localhost instance.
+    #>
+    $AllBackups = Backup-DbaDatabase -SqlInstance '____'
+    $AllBackups.ComputerName | Should -Be 'localhost'
```

Comment:
This is already pretty well broken up into blocks -- would be good to just wrap each block up as an `It` block so they register as separate koans. 🙂

Reply:
This file has the It blocks done